### PR TITLE
Workaround for cluster cleanup issue in end-to-end tests

### PR DESCRIPTION
**Pull request by khernyo**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| Related tickets | #643 
| License         | Apache 2.0

### What's in this PR?

Several small end-to-end test fixes/cleanups, and a "workaround" for resource cleanup after deletion of Istio resources. The "workaround" is to use an older kubernetes version for end-to-end testing, which does clean up all resources.

### Checklist

- [x] Implementation tested
- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [x] Logging code meets the guideline
- [x] User guide and development docs updated (if needed)

### CODE REVIEW

File: test/e2e/gvrs.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import "k8s.io/apimachinery/pkg/runtime/schema"
+
+var serviceGVR = schema.GroupVersionResource{
```
Comment (Laci21) [inline test/e2e/gvrs.go]: Aren't these defined somewhere already maybe?

**Review by Laci21 — APPROVED** [review]

File: test/e2e/gvrs.go (hunk 1 above)
Comment (khernyo) [reply test/e2e/gvrs.go]: I couldn't find any such definitions, but there are duplicated GVRs in istio-operator, so I'm gonna clean this up now

File: test/e2e/istio_controller_test_helpers.go [hunk 2]
```diff
@@ -341,117 +332,65 @@ func HPAExists(ctx context.Context, t *testing.T, kubeClient client.Client, name
...
+		gvr.MutatingWebhookConfiguration,
+		gvr.DestinationRule,
+		gvr.VirtualService,
+		gvr.PeerAuthentication,
+		gvr.Gateway,
+		gvr.EnvoyFilter,
+		gvr.AttributeManifest,
+		gvr.IstioConfigHandler,
+		gvr.IstioConfigInstance,
+		gvr.IstioConfigRule,
```
Comment (Laci21) [inline test/e2e/istio_controller_test_helpers.go]: I think these are probably Mixer related resources, I don't think we need these anymore.

Comment (khernyo) [reply test/e2e/istio_controller_test_helpers.go]: Yeah, that's what I was thinking too. Shouldn't we just rip the mixer related code out?

Comment (Laci21) [reply test/e2e/istio_controller_test_helpers.go]: Actually, I wouldn't spend a minute on that. Mixer won't be in v2, that should be enough. I'd just delete these from here.

Comment (khernyo) [reply test/e2e/istio_controller_test_helpers.go]: Yeah, that makes sense.

**Review by Laci21 — APPROVED** [review]
lgtm

**Review by waynz0r — APPROVED** [review]

**Pull request merged**